app/src/main/java/com/android/voyageur/model/user/UserViewModel.kt

```diff
@@ -579,39 +602,123 @@ open class UserViewModel(
-              // Fetch associated user data for notifications
-              getUsersByIds(requests.map { it.from }) { users -> _notificationUsers.value = users }
+
+              getUsersByIds(requests.map { it.from }) { users ->
+                _notificationUsers.value = users
+
+                // Check if there's a new request (list got bigger)
+                if (requests.size > oldRequests.size) {
+                  val oldRequestIds = oldRequests.map { it.id }.toSet()
+                  val newRequests = requests.filterNot { oldRequestIds.contains(it.id) }
+
+                  if (newRequests.isNotEmpty()) {
+                    val newRequest = newRequests.first()
+                    val senderUser = users.find { it.id == newRequest.from }
+                    val senderName = senderUser?.name ?: context?.getString(R.string.unknown)
+
+                    // Existing notification for a new friend request
+                    if (context != null) {
+                      NotificationHelper.showNotification(
+                          context = context,
+                          notificationId = 1001,
+                          title = context.getString(R.string.new_friend_request),
+                          text = context.getString(R.string.friend_request_message, senderName),
+                          iconResId = R.drawable.app_logo,
+                          priority = NotificationCompat.PRIORITY_HIGH)
+                    }
+                  }
+                }
+              }
             },
             onFailure = { exception ->
               Log.e("USER_VIEW_MODEL", "Failed to listen to friend requests: ${exception.message}")
             })
   }
+
+  /**
+   * Sets up a listener for sent friend requests so that the UI updates in real-time. This method
+   * also triggers a notification if a sent friend request is accepted.
+   */
+  fun getSentFriendRequests() {
+    val userId = Firebase.auth.uid.orEmpty()
+    if (userId.isEmpty()) return
+
+    val oldSentRequests = _sentFriendRequests.value
+    sentFriendRequestsListener?.remove()
+    sentFriendRequestsListener =
+        friendRequestRepository.listenToSentFriendRequests(
+            userId = userId,
+            onSuccess = { requests ->
+              val oldRequestsMap = oldSentRequests.associateBy { it.id }
+              _sentFriendRequests.value = requests
+
+              // Find requests that have just transitioned to accepted = true
+              val newlyAccepted =
+                  requests.filter { newReq ->
+                    newReq.accepted && oldRequestsMap[newReq.id]?.accepted == false
+                  }
+
+              if (newlyAccepted.isNotEmpty()) {
+                newlyAccepted.forEach { acceptedReq ->
+                  // The "to" user is the one who accepted
+                  getUsersByIds(listOf(acceptedReq.to)) { users ->
+                    val acceptorUser = users.firstOrNull()
```

Comment:
Here you should use the `getUserById` with `acceptedReq.to` as the query by index (as the ID is used as the direct key) is way faster than the `arrayContains` function used in the `getUsersByIds`

Reply:
I so not believe that getUserById is implemented. Am I missing something?

Reply:
Thank for the suggestion, you are absolutely right.